client/src/app/global-components/accounts/accounts.component.ts

```diff
@@ -0,0 +1,118 @@
+import { FormBuilder, Validators } from '@angular/forms';
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+
+import { Account } from './account.model';
+import { Currency } from './currency.modal';
+
+import { DataService } from '../services/data.service';
+import { AccountsService } from '../services/accounts.service';
+import { CurrenciesService } from '../services/currencies.service';
+
+@Component({
+  selector: 'app-accounts',
+  templateUrl: './accounts.component.html',
+  styleUrls: ['./accounts.component.scss']
+})
+
+export class AccountsComponent implements OnInit{
+
+  account: any = {};
+  accountId!: number;
+  accounts: Account[] = [];
+  currencies: Currency[] = [];
+  isOpenAddAccountModal: boolean = false;
+  isOpenViewAccountModal: boolean = false;
+  isCreateAccountError: boolean = false;
+  createAccountErrorMessage!: string;
+  toasterMessageStatus: boolean = false;
+  toasterMessageText!: string;
+
+  addAccountForm = this.fb.group({
+    title: ['', [Validators.required, Validators.maxLength(128)]],
+    currency: [0],
+    description: ['', [Validators.maxLength(256)]]
+  })
+
+  constructor(
+    private fb: FormBuilder,
+    private dataservice: DataService,
+    private accountsService: AccountsService,
+    private currenciesService: CurrenciesService,
+  ) { }
+
+  ngOnInit() {
+    this.getAccounts();
+    this.getCurrencies();
+    this.dataservice.currentaccountModalStatus.subscribe((status) => this.isOpenAddAccountModal = status)
+  }
+
+  getAccounts() {
+    this.accountsService.getAccounts().subscribe({
```

Comment:
Unsubscribe on destroy or use async pipe

Reply:
I use @ngneat/until-destroy for unsubscribe because it is a neat way